### PR TITLE
Unitwise metrics

### LINKED ISSUE #427: Generalize cluster quality metrics
The [clustering container](https://pynwb.readthedocs.io/en/latest/pynwb.ecephys.html#pynwb.ecephys.Clustering) has a dataset called "peak_over_rms". This is a metric of the quality of clustered units. Such metrics will likely be produced by every extracellular ephys experiment, but will not necessarily be peak_over_rms.  The peak_over_rms field should probably be replaced by a more generic quality metric field.

I will take a stab at this if there are no objections.

### PR DESCRIPTION
**Pull request by NileGraddis**
Resolves #427 

Adds a generic class for unit-wise metrics (such as the mean firing rate of a sorted ephys unit) and a specialization for peak_over_rms (the existing version of a unit metric). Adds a Multicontainer Interface for storing these metrics, which also stores a link to a unittimes group. Adds unit and integration tests for theses features

### CODE REVIEW

File: src/pynwb/data/nwb.misc.yaml [hunk 1]
```diff
@@ -139,3 +139,38 @@ groups:
...
+    - name: unit_times
+      target_type: UnitTimes
+      doc: defines the units from which these metrics were computed
+      quantity: 1
+  datasets: 
+    - neurodata_type_inc: UnitMetric
+      doc: a metric
+      quantity: '*'
+datasets:
+- neurodata_type_def: UnitMetric
```
Comment (ajtritt) [inline src/pynwb/data/nwb.misc.yaml]: Can we add a required description field here? If we allow for arbitrary metrics, we should also require that users provide a description of the metric they are storing.

**Review by ajtritt — CHANGES_REQUESTED** [review]

**Comment by tjd2002** [comment]
Per discussion with @NileGraddis it seems like a good idea to expand this PR to also include the proposed Unit class discussed at #431 . While this class would stand alone it seems preferable to refactor all the Unit-wise stuff together.

File: src/pynwb/data/nwb.misc.yaml (hunk 1 above)
Comment (tjd2002) [reply src/pynwb/data/nwb.misc.yaml]: Agreed

**Comment by bendichter** [comment]
@ajtritt can we close this?

**Pull request closed without merging**